dora/src/bytecode/writer.rs

```diff
@@ -20,6 +22,9 @@ pub struct BytecodeWriter {
 
     registers: Vec<BytecodeType>,
     const_pool: Vec<ConstPoolEntry>,
+
+    positions: Vec<(u32, Position)>,
+    position: Position,
```

Comment:
Wouldn't it be better if this is `Option<Position>`?

Reply:
The position gets set immediatly after the first statement is visited. After that, we never get `None`. Therefore i chose to avoid have two branches in `emit_position`, when we more or less never need the second branch.
But I guess it depends on personal preferences.

Reply:
There might be other users of BytecodeWriter in the future that don't want/need to record positions in the future (e.g. code generation). This would only affect `emit_position` (which is 3 lines long), so I think this is worth it.

Reply:
Done